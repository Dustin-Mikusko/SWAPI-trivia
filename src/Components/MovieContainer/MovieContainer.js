import React from 'react';
import './MovieContainer.css';
import Movie from '../Movie/Movie'

const MovieContainer = ({ movies }) => {
  const allMovies = movies.map((movie, i) => {
    return (
      < Movie
        key = {i}
        title = {movie.title}
        episode = {movie.episode}
        releaseDate = {movie.releaseDate}
        image = {movie.image}
        openingCredits = {movie.openingCredits}
      />
    )
  })

  return (
    <main>
    <h1>are you there?????</h1>
      {allMovies}
    </main>
  )
}

export default MovieContainer;
