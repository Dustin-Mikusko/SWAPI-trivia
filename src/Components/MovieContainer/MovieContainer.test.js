import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from './MovieContainer';

describe('MovieContainer', () => {

  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow (<MovieContainer
      movies = {
        [{
          key: 4,
          title: 'A Tale of Dustin Mikusko',
          episode: 1000,
          releaseDate: 1000,
          image: 'dustinisthebestpartnerever.com',
          openingCredits: 'still the best'
        }, {
          key: 44,
          title: 'A Tale of Trisha too',
          episode: 10,
          releaseDate: 2000,
          image: 'pants.com',
          openingCredits: 'not shorts'
        }]
      }
    />)
    expect(wrapper).toMatchSnapshot();
  })
})
