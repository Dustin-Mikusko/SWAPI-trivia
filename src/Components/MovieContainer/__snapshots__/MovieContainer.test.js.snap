// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should correctly match the snapshot with all the data passed correctly 1`] = `
<main>
  <[object Object]
    episode={1000}
    image="dustinisthebestpartnerever.com"
    key="0"
    openingCredits="still the best"
    releaseDate={1000}
    title="A Tale of Dustin Mikusko"
  />
  <[object Object]
    episode={10}
    image="pants.com"
    key="1"
    openingCredits="not shorts"
    releaseDate={2000}
    title="A Tale of Trisha too"
  />
</main>
`;
